lib/skate/detours/route_segments.ex

```diff
@@ -16,8 +27,28 @@ defmodule Skate.Detours.RouteSegments do
     defstruct [:before_detour, :detour, :after_detour]
   end
 
+  @spec unfinished_route_segments(
+          nonempty_list(Util.Location.From.t()),
+          Util.Location.From.t()
+        ) :: {:ok, UnfinishedResult.t()}
+  def unfinished_route_segments([], _start_point), do: :error
+  def unfinished_route_segments([_], _start_point), do: :error
+
+  def unfinished_route_segments(
+        shape,
+        start_point
+      ) do
+    {nearest_point, index} = NearestPoint.nearest_point_on_shape(shape, start_point)
+
+    {:ok,
+     %__MODULE__.UnfinishedResult{
+       before_start_point: Enum.slice(shape, 0..index) ++ [nearest_point],
+       after_start_point: [nearest_point] ++ Enum.slice(shape, (index + 1)..-1)
+     }}
```

Comment:
question(non-blocking): I find the overhead of understanding this _slightly_ annoying, could we use `Enum.split` instead?

Something like this?
```suggestion
    [shape_before_start_point, shape_after_start_point] = Enum.split(shape, index)

    {:ok,
     %__MODULE__.UnfinishedResult{
       before_start_point: shape_before_start_point ++ [nearest_point],
       after_start_point: [nearest_point] ++ shape_after_start_point
     }}
```

Reply:
Oh huh - I assumed that `Enum.split/2` would work more similarly to [`String.split/3`](https://hexdocs.pm/elixir/1.12/String.html#split/3), so it didn't even occur to me to do it this way.

But unless I'm missing something, your way is a lot cleaner, both for this and for the other route segments function - will do a follow-up

Reply:
https://hexdocs.pm/elixir/1.17.0-rc.1/enum-cheat.html has been very helpful for me fwiw.